src/components/marker-chart/Canvas.js

```diff
@@ -114,9 +118,27 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
           ? undefined
           : markerIndexToTimingRow.get(hoveredItem);
 
-      if (newRow !== undefined) {
+      if (newRow !== undefined && newRow !== rightClickedRow) {
+        /**
+         * To avoid multiple renders on the same row rectangle in canvas, first the row
+         * is cleared and then redrawn on using the drawMarkers()
+         **/
+        ctx.clearRect(
+          0, // To include the labels also
+          newRow * rowHeight - viewportTop,
+          containerWidth,
+          rowHeight - 1 // Subtract 1 for borders.
+        );
         this.clearRow(ctx, newRow);
         this.drawMarkers(ctx, hoveredItem, newRow, newRow + 1);
+        // To highlight the row when mouse hovered over
+        ctx.fillStyle = 'rgba(40, 122, 169, 0.2)';
+        ctx.fillRect(
+          0, // To include the labels also
+          newRow * rowHeight - viewportTop,
+          containerWidth,
+          rowHeight - 1 // Subtract 1 for borders.
+        );
```

Comment:
It's working either way. Are there any performance implications tho?

Reply:
Here it's not for the performance but for the effect => I think that drawing markers afterwards makes them more visible, more contrasted. Because if you draw them before, the highlight will  go on top of them and wash them out.